src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[] dataOne;
+        int [] namesSalary = new int[names.length];
+        for (String personData : data) {
+            dataOne = personData.split(" ");
+            if (compareDate(getLocalDate(dataOne[0]),getLocalDate(dateFrom),getLocalDate(dateTo))) {
+                for (int i = 0; i < names.length; i++) {
+                    if (dataOne[1].equals(names[i])) {
+                        namesSalary[i] += Integer.parseInt(dataOne[3])
+                                * Integer.parseInt(dataOne[2]);
+                    }
+                }
+            }
+        }
+        return getResultNameAndSalary(names,namesSalary,dateFrom,dateTo).toString();
+    }
+
+    public StringBuilder getResultNameAndSalary(String[] names,
```

Comment:
let's not return StringBuilder, you can simply return String representation. I'd recommend renaming it to 'createReport' or something like this. Also, using two different arrays names and namesSalary is not a good practice because of possible inconsistent system states. It will be better to introduce a new class e.g. Employee with fields name and salary . And use one Employee[] employees array for passing data instead of two different arrays.